internal/innerConcat.js

```diff
@@ -0,0 +1,26 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const curry = require('../helpers/curry')
+const isSameType = require('../predicates/isSameType')
+const isSemigroup = require('../predicates/isSemigroup')
+
+function innerConcat(type, m) {
+  const t = type.type()
+
+  return function(left) {
+    if(!isSemigroup(left)) {
+      throw new TypeError(`${t}.concat: Both containers must contain Semigroups of the same type`)
+    }
+
+    return m.map(right => {
+      if(!isSameType(left, right)) {
+        throw new TypeError(`${t}.concat: Both containers must contain Semigroups of the same type`)
+      }
```

Comment:
From gitter chat, it seems that you meant that I should be able to [ 'a' ].concat('b'). While that is true in JS, for these types of interaction I am going to go with the FL spec for semigroup.

So I do not see an error here.